nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateJson.java

```diff
@@ -0,0 +1,202 @@
+        .name("valid")
+        .description("FlowFiles that are successfully validated against the schema are routed to this relationship")
+        .build();
+
+    public static final Relationship REL_INVALID = new Relationship.Builder()
+        .name("invalid")
+        .description("FlowFiles that are not valid according to the specified schema are routed to this relationship")
+        .build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+        .name("failure")
+        .description("FlowFiles that cannot be read as JSON are routed to this relationship")
+        .build();
+
+    private List<PropertyDescriptor> descriptors;
+
+    private Set<Relationship> relationships;
+
+    private List<AllowableValue> allowableValues;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> descriptors = new ArrayList<PropertyDescriptor>();
+        descriptors.add(SCHEMA_TEXT);
+        descriptors.add(SCHEMA_VERSION);
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<Relationship>();
+        relationships.add(REL_VALID);
+        relationships.add(REL_INVALID);
+        relationships.add(REL_FAILURE);
+        this.relationships = Collections.unmodifiableSet(relationships);
+
+        final List<AllowableValue> allowableValues = new ArrayList<AllowableValue>();
+        allowableValues.add(SCHEMA_VERSION_4);
+        allowableValues.add(SCHEMA_VERSION_6);
+        allowableValues.add(SCHEMA_VERSION_7);
+        allowableValues.add(SCHEMA_VERSION_V201909);
+        this.allowableValues = Collections.unmodifiableList(allowableValues);
+
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    private ObjectMapper mapper = new ObjectMapper();
+    private VersionFlag schemaVersion;
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        // Set JSON schema version to use from processor property
+        this.schemaVersion = VersionFlag.V201909;
+        if (context.getProperty(SCHEMA_VERSION).getValue() == SCHEMA_VERSION_4.getValue()) {
+            this.schemaVersion = VersionFlag.V4;
+        }
+        if (context.getProperty(SCHEMA_VERSION).getValue() == SCHEMA_VERSION_6.getValue()) {
+            this.schemaVersion = VersionFlag.V6;
+        }
+        if (context.getProperty(SCHEMA_VERSION).getValue() == SCHEMA_VERSION_7.getValue()) {
+            this.schemaVersion = VersionFlag.V7;
+        }
+        if (context.getProperty(SCHEMA_VERSION).getValue() == SCHEMA_VERSION_V201909.getValue()) {
+            this.schemaVersion = VersionFlag.V201909;
+        }
```

Comment:
It would be helpful to refactor this approach to use a separate Java `enum`, which would support parsing the property instead of including these conditionals.  The `enum` should implement `DescribedValue`, which will allow using it for the `allowableValues` in the Property Descriptor. The `enum` can also have a `getVersionFlag()` method to return the JSON Schema VersionFlag value.